llama-stack-client-kotlin-client-local/src/main/kotlin/com/llama/llamastack/client/local/InferenceServiceLocalImpl.kt

```diff
@@ -0,0 +1,104 @@
+// File generated from our OpenAPI spec by Stainless.
+
+package com.llama.llamastack.client.local
+
+import com.llama.llamastack.client.local.util.PromptFormatLocal
+import com.llama.llamastack.core.JsonValue
+import com.llama.llamastack.core.RequestOptions
+import com.llama.llamastack.models.CompletionMessage
+import com.llama.llamastack.models.EmbeddingsResponse
+import com.llama.llamastack.models.InferenceChatCompletionParams
+import com.llama.llamastack.models.InferenceChatCompletionResponse
+import com.llama.llamastack.models.InferenceCompletionParams
+import com.llama.llamastack.models.InferenceCompletionResponse
+import com.llama.llamastack.models.InferenceEmbeddingsParams
+import com.llama.llamastack.services.blocking.InferenceService
+import org.pytorch.executorch.LlamaCallback
+
+class InferenceServiceLocalImpl
+constructor(
+    private val clientOptions: LocalClientOptions,
+) : InferenceService, LlamaCallback {
+
+    private var resultMessage: String = ""
+    private var onResultComplete: Boolean = false
+    private var statsMetric: Float = 0.0f
+    private var onStatsComplete: Boolean = false
+    private var modelName: String = ""
+
+    private var sequenceLengthKey: String = "seq_len"
+
+    override fun onResult(p0: String?) {
+        if (p0.equals(PromptFormatLocal.getStopToken(modelName))) {
+            onResultComplete = true
+            return
+        }
+
+        if (p0.equals("\n\n") || p0.equals("\n")) {
+            if (resultMessage.isNotEmpty()) {
+                resultMessage += p0
+            }
+        } else {
+            resultMessage += p0
+        }
+    }
+
+    override fun onStats(p0: Float) {
+        onResultComplete =
+            true // required since in some cases where seq_len is met then EOT is not appended by ET
+        // logic
+        statsMetric = p0
+        onStatsComplete = true
+    }
+
+    override fun chatCompletion(
+        params: InferenceChatCompletionParams,
+        requestOptions: RequestOptions
+    ): InferenceChatCompletionResponse {
+        resultMessage = ""
+        val mModule = clientOptions.llamaModule
+        modelName = params.modelId()
+        val formattedPrompt =
+            PromptFormatLocal.getTotalFormattedPrompt(params.messages(), modelName)
+        val seqLength =
+            params._additionalQueryParams().values(sequenceLengthKey).lastOrNull()?.toInt()
+                ?: ((formattedPrompt.length * 0.75) + 64).toInt()
```

Comment:
What? (the numbers)

Reply:
We did this similar as well with the ExecuTorch demo app (https://github.com/pytorch/executorch/blob/main/examples/demo-apps/android/LlamaDemo/app/src/main/java/com/example/executorchllamademo/MainActivity.java#L758). The idea here is to make sure that enough sequence length value is provided into the function such that it will provide a reasonably complete response.

0.75 is the approximate words per token. And 64 is buffer for tokens for generate response. 

At the same time, we have seen some behaviors where ET library doesn't necessarily acknowledge the exact sequence length so we have a task for this. However we preserved this logic since it works well in ET demo app.

Reply:
@cmodi-meta If we want to keep this hardcoded number, can you add explanation inline?

Reply:
makes sense! added comments to explain inline.